### PR TITLE
[mlir][linalg][nfc] Delete references to args_in/args_out

### PR DESCRIPTION
**Pull request by banach-space**
After the refactor in:
  * ed229132f1c4ea2ba0644fc345d8279e47a00565,

the `args_in` and `args_out` attributes are no longer used by
`linalg.generic`. This patch removes all the remaining references.

### CODE REVIEW

File: mlir/docs/BufferDeallocationInternals.md [hunk 1]
```diff
@@ -664,8 +664,6 @@ merged into a single step. Canonicalization removes the clone operation and
 func.func @reuseTarget(%arg0: memref<2xf32>, %result: memref<2xf32>){
   %temp = memref.alloc() : memref<2xf32>
   test.generic {
-    args_in = 1 : i64,
-    args_out = 1 : i64,
```
Comment (javedabsar1) [inline mlir/docs/BufferDeallocationInternals.md:668]: Just deleting will still leave IR incorrect. The IR below should be like (Notice the ins , outs and math.exp, and braces ...

```
    %alloc = memref.alloc() : memref<2xf32>
    linalg.generic {indexing_maps = [#map, #map], iterator_types = ["parallel"]} ins(%arg1 : memref<2xf32>) outs(%alloc : memref<2xf32>) {
    ^bb0(%in: f32, %out: f32):
      %1 = math.exp %in : f32
      linalg.yield %1 : f32
    }
``

**Review by javedabsar1 — CHANGES_REQUESTED** [review]
Thanks @banach-space for spotting these. Some comments on getting it totally fixed.

File: mlir/docs/BufferDeallocationInternals.md (hunk 1 above)
Comment (banach-space) [reply mlir/docs/BufferDeallocationInternals.md:668]: Thanks for spotting that. This example is broken in other ways too, e.g (the parser rejects this):
```mlir
%result = bufferization.clone %temp : (memref<2xf32>) -> (memref<2xf32>)
```

Also, at the top of the file:
```
**Note:** This pass is deprecated. Please use the ownership-based buffer
deallocation pass instead.
```

I feel that I should skip this file altogether, it clearly hasn't been maintained for a while now.

Comment (javedabsar1) [reply mlir/docs/BufferDeallocationInternals.md]: Even if you don’t fix everything … fixing some parts is already an improvement .

Comment (banach-space) [reply mlir/docs/BufferDeallocationInternals.md]: What are you suggesting? Partial update in this file? TBH, I'd rather skip it.

Comment (javedabsar1) [reply mlir/docs/BufferDeallocationInternals.md]: Up to you really.
 I meant if args_in args_out are present in files like `mlir/include/mlir/Dialect/Bufferization/Transforms/Passes.td` which lot of folks read, removing  it is still better than 'fix everything or fix nothing'
then

Comment (banach-space) [reply mlir/docs/BufferDeallocationInternals.md]: Sounds like I've addressed your comment then? Asking as according to GitHub you are still "requesting changes" :)

**Review by javedabsar1 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**